src/ReverseProxy/Configuration/ConfigurationConfigProvider.cs

```
@@ -120,225 +127,247 @@ private void UpdateSnapshot(ConfigurationData data)
 
             return new QuotaOptions
             {
-                Average = data.Average,
-                Burst = data.Burst,
+                Average = double.TryParse(section[nameof(QuotaOptions.Average)], out var avg) ? avg : default,
+                Burst = double.TryParse(section[nameof(QuotaOptions.Burst)], out var burst) ? burst : default,
             };
         }
 
-        private static ClusterPartitioningOptions Convert(ClusterPartitioningData data)
+        private static ClusterPartitioningOptions CreateClusterPartitioningOptions(IConfigurationSection section)
         {
-            if (data == null)
+            if (section?.Exists() is false)
             {
                 return null;
             }
 
             return new ClusterPartitioningOptions
             {
-                PartitionCount = data.PartitionCount,
-                PartitionKeyExtractor = data.PartitionKeyExtractor,
-                PartitioningAlgorithm = data.PartitioningAlgorithm,
+                PartitionCount = int.TryParse(section[nameof(ClusterPartitioningOptions.PartitionCount)], out var partitionCount) ? partitionCount : default,
+                PartitionKeyExtractor = section[nameof(ClusterPartitioningOptions.PartitionKeyExtractor)],
+                PartitioningAlgorithm = section[nameof(ClusterPartitioningOptions.PartitioningAlgorithm)],
             };
         }
 
-        private static LoadBalancingOptions Convert(LoadBalancingData data)
+        private static LoadBalancingOptions CreateLoadBalancingOptions(IConfigurationSection section)
         {
-            if (data == null)
+            if (section?.Exists() is false)
             {
                 return null;
             }
 
             return new LoadBalancingOptions
             {
-                Mode = Enum.Parse<LoadBalancingMode>(data.Mode),
+                Mode = Enum.TryParse<LoadBalancingMode>(section[nameof(LoadBalancingOptions.Mode)], out var mode) ? mode : default,
```

Should we use a more aggressive failure policy than TryParse? E.g. Parse? We already throw exceptions if we can't load certificates, why not fail if we can't parse the int's, enums, etc? Silent failures here seem impossible to troubleshoot.

---

I don't have an issue with that. I think I initially had that and ended up with regressions but I can do it again. Only parse if not-null.